adapters/oidc/adapter-core/src/main/java/org/keycloak/adapters/OAuthRequestAuthenticator.java

```diff
@@ -387,22 +387,22 @@ protected String stripOauthParametersFromRedirect() {
                 .replaceQueryParam(OAuth2Constants.SESSION_STATE, null);
         return builder.build().toString();
     }
-    
+
     private String rewrittenRedirectUri(String originalUri) {
         Map<String, String> rewriteRules = deployment.getRedirectRewriteRules();
-            if(rewriteRules != null && !rewriteRules.isEmpty()) {
+        if(rewriteRules != null && !rewriteRules.isEmpty()) {
```

Comment:
You can use Apache Commons for check that map == null and is empty
```
CollectionUtils.isNotEmpty(rewriteRules)
```

Reply:
I pushed a fixup with `MapUtils.isNotEmpty(rewriteRules)`

Reply:
Thinking my code over I came up with

`URI uri = new URI(originalUrl);
return new URI(uri.getScheme(),
                          uri.getAuthority(),
			  uri.getPath().replaceFirst(rule.getKey(), rule.getValue()),
			  uri.getQuery(),
			  uri.getFragment()).toString();`

which (maybe) has less overhead than my first solution.